hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java

```diff
@@ -789,6 +789,15 @@ public OmKeyInfo copyObject() {
     return builder.build();
   }
 
+  /**
+   * Return a new copy of the object.
+   */
+  @Override
+  public Object clone() throws CloneNotSupportedException {
+    OmKeyInfo omKeyInfo = (OmKeyInfo) super.clone();
+    return omKeyInfo.copyObject();
```

Comment:
Here clone creates a new reference but with shallowcopy but when moving to copyObject->build() again a new OmkeyInfo object is created to enable deep copy. Though ideally you needed deep copy using same reference returned by clone. Hope this behavior is appropriate.

Reply:
Handled the change.